src/transport.rs

```diff
@@ -115,6 +116,7 @@ pub struct ServerConnection {
 
     channels: Option<MessageChannels>,
     channels_rx: Option<IncomingMultiplexedPackets<MultiplexedPacket>>,
+    #[allow(dead_code)]
```

Comment:
Why is this needed?

Reply:
to silence this warning:

```
warning: field is never read: `channels_task`
   --> src/transport.rs:121:5
    |
121 |     channels_task: Option<Task<()>>,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default

warning: 1 warning emitted
```

there is a spawned task assigned to it in build_channels, but we never read from it. looks like that's just to keep the task in scope so it's not dropped. is there a better way to make the compiler happy?

Reply:
Oh yeah. Right.

I wish it was a way to explicitly mark this type of use, as `dead_code` suggests that this can be removed.
We need to add a comment explaining that this field is used for reference keeping and cannot be removed.

Reply:
will do